pkg/kvstore/events.go

```diff
@@ -75,16 +81,27 @@ type Watcher struct {
 	prefix    string
 	stopWatch stopChan
 
+	// mutex protects access to stopped
+	mutex lock.Mutex
+
+	// stopped is true when the watcher has been stopped
 	stopped bool
+
+	// stopWait is the wait group to wait for watchers to exit gracefully
+	stopWait sync.WaitGroup
```

Comment:
Both `stopWait` and `stopped` could be replaced with a single channel (e.g. named `stopped`) which would be closed by the watcher on exit. `Stop` could become:
```
w.mutex.Lock()
select {
    case <-w.stopWatch: // already stopped
        return
    default:
        close(w.stopWatch)
}
w.mutex.Unlock()
log.WithField(fieldWatcher, w).Debug("Stopped watcher")
<-w.stopped
```

Reply:
Nice, I like it. Adjusted accordingly.